com.dynamo.cr/com.dynamo.cr.bob/src/com/defold/extension/pipeline/texture/ITextureCompressor.java

```diff
@@ -23,4 +24,8 @@ public interface ITextureCompressor {
     public String getName();
 
     public byte[] compress(TextureCompressorPreset preset, TextureCompressorParams params, byte[] input);
+
+    public default int getAlignedWidth(TextureImage.TextureFormat format, int width) { return width; }
+
+    public default int getAlignedHeight(TextureImage.TextureFormat format, int height) { return height; }
```

Comment:
An alternative to changing the API would be to check specifically in the texture compressor if we are using an ASTC format, but at least this lets the plugins decide themselves how to calculate the alignment.